packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -355,13 +355,7 @@ class RenderParagraph extends RenderBox {
 
     if (_hasVisualOverflow) {
       final Rect bounds = offset & size;
-      if (_overflowShader != null) {
-        // This layer limits what the shader below blends with to be just the text
-        // (as opposed to the text and its background).
-        canvas.saveLayer(bounds, new Paint());
-      } else {
         canvas.save();
-      }
```

Comment:
this sounds like it breaks the rendering, is there a test that verifies that this is ok?

Reply:
ah, that's https://github.com/flutter/flutter/issues/18729
please add a golden test when fixing this. thanks.